pandas/tools/merge.py

```diff
@@ -1436,13 +1438,198 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
-    -----
-    The keys, levels, and names arguments are all optional
-
     Returns
     -------
     concatenated : type of objects
+
+    Notes
+    -----
+    The keys, levels, and names arguments are all optional.
+
+    A walkthrough of how this method fits in with other tools for combining
+    panda objects can be found `here </pandas-docs/stable/merging.html>`_.
+
+    See Also
+    --------
+    Series.append
+    DataFrame.append
+    DataFrame.join
+    DataFrame.merge
+    Panel.join
+
+    Examples
+    --------
+    Combine two ``Series``.
+
+    >>> s1 = pd.Series(['a', 'b'])
+    >>> s2 = pd.Series(['c', 'd'])
+    >>> pd.concat([s1, s2])
+    0    a
+    1    b
+    0    c
+    1    d
+    dtype: object
+
+    Clear the existing index and reset it in the result
+    by setting the ``ignore_index`` option to ``True``.
+
+    >>> s1 = pd.Series(['a', 'b'])
+    >>> s2 = pd.Series(['c', 'd'])
+    >>> pd.concat([s1, s2], ignore_index=True)
+    0    a
+    1    b
+    2    c
+    3    d
+    dtype: object
+
+    Add a hierarchical index at the outermost level of
+    the data with the ``keys`` option.
+
+    >>> s1 = pd.Series(['a', 'b', 'c'])
+    >>> s2 = pd.Series(['c', 'd', 'e'])
+    >>> pd.concat([s1, s2], keys=['s1', 's2',])
+    s1  0    a
+        1    b
+        2    c
+    s2  0    c
+        1    d
+        2    e
+    dtype: object
+
+    Label the index keys you create with the ``names`` option.
+
+    >>> s1 = pd.Series(['a', 'b', 'c'])
+    >>> s2 = pd.Series(['c', 'd', 'e'])
+    >>> pd.concat(
+    ...     [s1, s2],
+    ...     keys=['s1', 's2',],
+    ...     names=['Series name', 'Row ID']
+    ... )
+    Series name  Row ID
+    s1           0         a
+                 1         b
+                 2         c
+    s2           0         c
+                 1         d
+                 2         e
+    dtype: object
+
+    Combine two ``DataFrame`` objects with identical columns.
+
+    >>> df1 = pd.DataFrame(
+    ...     [['a', 1], ['b', 2]],
+    ...     columns=['letter', 'number']
+    ... )
+    >>> df1
+      letter  number
+    0      a       1
+    1      b       2
+    >>> df2 = pd.DataFrame(
+    ...     [['c', 3], ['d', 4]],
+    ...     columns=['letter', 'number']
+    ... )
+    >>> df2
+      letter  number
+    0      c       3
+    1      d       4
+    >>> pd.concat([df1, df2])
+      letter  number
+    0      a       1
+    1      b       2
+    0      c       3
+    1      d       4
+
+    Combine ``DataFrame`` objects with overlapping columns
+    and return everything. Columns outside the intersection will
+    be filled with ``NaN`` values.
+
+    >>> df1 = pd.DataFrame(
+    ...     [['a', 1], ['b', 2]],
+    ...     columns=['letter', 'number']
+    ... )
+    >>> df1
+      letter  number
+    0      a       1
+    1      b       2
+    >>> df2 = pd.DataFrame(
+    ...     [['c', 3, 'cat'], ['d', 4, 'dog']],
+    ...     columns=['letter', 'number', 'animal']
+    ... )
```

Comment:
I meant this comment actually for the example below, as here df2 is actually changed (maybe only show df2 creation and display in this example)